packages/block-editor/src/components/spacing-sizes-control/spacing-input-control.js

```diff
@@ -227,7 +229,16 @@ export default function SpacingInputControl( {
 					<RangeControl
 						value={ customRangeValue }
 						min={ 0 }
-						max={ 300 }
+						max={
+							maxCustomValues[ selectedUnit ]
+								? maxCustomValues[ selectedUnit ]
+								: 10
+						}
+						step={
+							[ 'px', '%', 'vh', 'vw' ].includes( selectedUnit )
+								? 1
+								: 0.1
+						}
```

Comment:
Might be good to do this the same way `maxCustomValues` works, extract it out to a variable.